openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/ConfigurationDialog.java

```diff
@@ -93,4 +111,41 @@ public ConfigurationDialog() {
         add(panel);
     }
 
+    private void updateVendorComboBox(URL specifiedServerURL) {
+        try {
+            SwingUtilities.invokeLater(() -> vendorComboBox.setCursor(getPredefinedCursor(WAIT_CURSOR)));
+            final String[] vendorNames = JavaRuntimeManager.getAllVendors(specifiedServerURL);
+            SwingUtilities.invokeLater(() -> {
+                vendorComboBox.setModel(new DefaultComboBoxModel(vendorNames));
+                vendorComboBox.setSelectedItem(RuntimeManagerConfig.getVendor());
+            });
+        } catch (Exception ex) {
+            DialogFactory.showErrorDialog(translator.translate("jvmManager.error.updateVendorNames"), ex);
+        } finally {
+            SwingUtilities.invokeLater(() -> vendorComboBox.setCursor(getDefaultCursor()));
+        }
+    }
+
+    private static URL getUrl(String text) {
+        try {
+            return new URL(text);
+        } catch (MalformedURLException e) {
+            return null;
+        }
+    }
+
+    private class MyFocusAdapter extends FocusAdapter {
+        @Override
+        public void focusLost(FocusEvent e) {
+            final JTextField field = (JTextField) e.getSource();
```

Comment:
I assume that a cast to Component will be better since we do not need to be that concrete (JTextField). Next to this you should check the type before doing the cast.